src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js

```diff
@@ -185,6 +185,12 @@ function ComposerWithSuggestions({
             setIsCommentEmpty(!!newComment.match(/^(\s)*$/));
             setValue(newComment);
             if (commentValue !== newComment) {
+                // Reset emoji suggestions when an emoji is replaced.
+                // Important when emoji is replaced and the selection is not changed, more info issue #27156
+                if (suggestionsRef.current) {
```

Comment:
> I guess it's possible that in future newComment might not only be changed by emoji replacements

@jjcoffee 
I agree about that, but I think `!_.isEmpty(emojis)` will not be useful in this case because `emojis` it returns all emoji in the text not only the current replaced emojis.
 For example: `text with emoji 😄👁️ then many lines of text then cursor at the end writing normal letters not emoji` will return emojis = [😄, 👁️]
 

`newComment` now in the code refer to comment after replace emoji.
 So I think If new replacement applied in the future, in this time the fix should be `if (commentValueWithEmoji !== commentValueWithoutEmoji)`

Reply:
@ahmedGaber93 Ah yes I see what you mean, that makes sense. Let's leave it for now then!